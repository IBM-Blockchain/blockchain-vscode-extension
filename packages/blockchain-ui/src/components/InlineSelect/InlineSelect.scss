@import '@carbon/layout/scss/_spacing';

.inline-select {
    display: inline-flex;
    flex-direction: row;

    .bx--select-input__wrapper {
        margin-left: $spacing-03;
    }

    .bx--select-input {
        background: none;
        border: none;
    }

    .bx--label {
        margin-top: 11.5px;
        margin-bottom: 0px;
        font-size: 100%;
        color: inherit;
    }

}
