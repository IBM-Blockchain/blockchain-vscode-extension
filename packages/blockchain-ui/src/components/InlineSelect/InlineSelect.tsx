import React, { Component } from 'react';
import './InlineSelect.scss';
import { Select } from 'carbon-components-react';

interface InlineSelectProps {
    id: string;
    labelText: string;
    contents: Array<JSX.Element>;
    onChangeCallback: (event: React.FormEvent<HTMLSelectElement>) => void;
}

interface InlineSelectState {
    id: string;
    labelText: string;
    contents: Array<JSX.Element>;
    onChangeCallback: (event: React.FormEvent<HTMLSelectElement>) => void;
}

class InlineSelect extends Component<InlineSelectProps, InlineSelectState> {

    constructor(props: Readonly<InlineSelectProps>) {
        super(props);
        this.state = {
            id: this.props.id,
            labelText: this.props.labelText,
            contents: this.props.contents,
            onChangeCallback: this.props.onChangeCallback
        };
    }

    render(): JSX.Element {
        return (
            <div id={this.state.id + '-wrapper'}>
                <Select id={this.state.id} labelText={this.state.labelText} className='inline-select' onChange={this.state.onChangeCallback}>
                    {this.state.contents}
                </Select>
            </div>
        );
    }

}

export default InlineSelect;
