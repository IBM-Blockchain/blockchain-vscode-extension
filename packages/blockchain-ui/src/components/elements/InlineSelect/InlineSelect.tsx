import React, { Component } from 'react';
import './InlineSelect.scss';
import { Select } from 'carbon-components-react';

interface IProps {
    id: string;
    labelText: string;
    contents: Array<JSX.Element>;
    onChangeCallback: (event: React.FormEvent<HTMLSelectElement>) => void;
}

class InlineSelect extends Component<IProps> {

    constructor(props: Readonly<IProps>) {
        super(props);
        this.state = {
            id: this.props.id,
            labelText: this.props.labelText,
            contents: this.props.contents,
            onChangeCallback: this.props.onChangeCallback
        };
    }

    render(): JSX.Element {
        return (
            <div id={this.props.id + '-wrapper'}>
                <Select id={this.props.id} labelText={this.props.labelText} className='inline-select' onChange={this.props.onChangeCallback}>
                    {this.props.contents}
                </Select>
            </div>
        );
    }

}

export default InlineSelect;
