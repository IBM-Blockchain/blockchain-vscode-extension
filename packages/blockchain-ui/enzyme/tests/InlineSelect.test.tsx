import React from 'react';
import renderer from 'react-test-renderer';
import InlineSelect from '../../src/components/elements/InlineSelect/InlineSelect';
import { SelectItem } from 'carbon-components-react';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
chai.should();
chai.use(sinonChai);

describe('InlineSelect component', () => {

    let mySandBox: sinon.SinonSandbox;

    const mockOptions: Array<JSX.Element> = [
        <SelectItem disabled={false} hidden={false} text='Inline Select Option 1' value={'Inline Select Option 1'}/>,
        <SelectItem disabled={false} hidden={false} text='Inline Select Option 2' value={'Inline Select Option 2'}/>
    ];

    beforeEach(async () => {
        mySandBox = sinon.createSandbox();
    });

    afterEach(async () => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', () => {
        const onChangeStub: sinon.SinonStub = mySandBox.stub();
        const component: any = renderer
            .create(<InlineSelect id='my-inline-select' labelText='My Inline Select' contents={mockOptions} onChangeCallback={onChangeStub}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
