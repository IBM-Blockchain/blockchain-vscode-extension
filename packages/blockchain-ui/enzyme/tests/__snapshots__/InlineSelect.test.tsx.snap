// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineSelect component should render the expected snapshot 1`] = `
<div
  id="my-inline-select-wrapper"
>
  <div
    className="bx--form-item"
  >
    <div
      className="bx--select inline-select"
    >
      <label
        className="bx--label"
        htmlFor="my-inline-select"
      >
        My Inline Select
      </label>
      <div
        className="bx--select-input__wrapper"
        data-invalid={null}
      >
        <select
          className="bx--select-input"
          id="my-inline-select"
          onChange={[Function]}
        >
          <option
            className="bx--select-option"
            disabled={false}
            hidden={false}
            value="Inline Select Option 1"
          >
            Inline Select Option 1
          </option>
          <option
            className="bx--select-option"
            disabled={false}
            hidden={false}
            value="Inline Select Option 2"
          >
            Inline Select Option 2
          </option>
        </select>
        <svg
          aria-hidden={true}
          className="bx--select__arrow"
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="8,11 3,6 3.7,5.3 8,9.6 12.3,5.3 13,6"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
